### ISSUE TITLE
Set master address via flag

### ISSUE THREAD

**Issue by geoffgarside** [saltstack/salt-bootstrap#85]
I'm wondering if it would be worthwhile to be able to specify the address/name of the master when bootstrapping a minion as the daemons are now started as part of the bootstrapping process.

When bootstrapping a master with a minion perhaps the address/name could be defaulted to "localhost".

Thoughts?

**Comment by maru1034sk** [comment]
+1

Would be awesome.

**Comment by geoffgarside** [comment]
I'm thinking `-m` as the argument, I'll see if I can get this coded up at least at the weekend. Assuming the salt minion sample config is installed the same across all the supported operating systems it should be pretty easy to `sed` the `# master: salt` line, the fun will come with making the `sed` call portable due to the lovely differences between GNU and BSD sed :(

**Comment by s0undt3ch** [comment]
I should have replied to this sooner.

We can already solve this problem by uploading a properly tweaked minion configuration file, no?

**Comment by geoffgarside** [comment]
How would that be done? Is that a command line flag already?

**Comment by s0undt3ch** [comment]
The bootstrap script is prepared for looking for the minion and master keys along with their configurations, have a look [here](https://github.com/saltstack/salt-bootstrap/blob/develop/bootstrap-salt.sh#L1980).

This was added to improve support and functionality for [salt-cloud](https://github.com/saltstack/salt-cloud) and [salty-vagrant](https://github.com/saltstack/salty-vagrant).

In order to trigger this "look up" functionality:

```
sh bootstrap-salt.sh -c /path/to/directory/containing/the/required/files
```

The "upload" process is not handled by the bootstrap script.
I think salty-vagrant just copies the required files files into the VM, while salt-cloud uploads them using scp. ( I think :smile: )

**Comment by geoffgarside** [comment]
Ah ok, so the files have to exist prior to running boostrap-salt.sh, might be nice to specify an FTP directory to fetch them from or even HTTP if the file names are known beforehand.

One thing I've noticed from that though is the SALT_DIR is coded to /etc/salt, on BSD installations the SALT_DIR should be /usr/local/etc/salt as thats where the port is configured to use. I might have a stab then at adding http or ftp support to the `config_salt` function, would be really nice to be able to use the `salt://` type URLs but I think thats probably unlikely.

**Comment by s0undt3ch** [comment]
Yes, files must exist before hand.

About the remote directory, that would be cool, but note that it needs to be supported by all distros(this is the hard part).
Regarding having salt do it(`salt://`), this is the chicken and the egg question ;)

About the salt dir, we could fix that by allowing a user environment directory to be present, `BS_SALT_DIR`...

**Comment by s0undt3ch** [comment]
@geoffgarside want me to keep this ticket open?

**Comment by geoffgarside** [comment]
Yeah you can close it, I'll submit a pull request when I've got something useful :)

**Comment by s0undt3ch** [comment]
Cool!
Closed.

**Comment by chadhs** [comment]
for your bootstrap one–liner in your environment try adding this on:
`&& echo "master: yoursaltmaster.yourdomain.ext" >> /your/salt/config/path/minion && service salt_minion restart`

edit: once you do that you've got connectivity with the master and you can replace the minion file if you have/need other stuff in there and don't want the master definition to be at the bottom.  FYI

**Comment by s0undt3ch** [comment]
Could also work :smile:

**Comment by s0undt3ch** [comment]
This is being implemented in #307
